src/software/ai/navigator/path_planner/theta_star_path_planner.cpp

```diff
@@ -63,14 +63,25 @@ bool ThetaStarPathPlanner::lineOfSight(const Coordinate &coord1, const Coordinat
     auto line_of_sight_cache_it = line_of_sight_cache.find(coordinate_pair_key);
     if (line_of_sight_cache_it == line_of_sight_cache.end())
     {
-        Segment seg(convertCoordToPoint(coord1), convertCoordToPoint(coord2));
-        bool has_line_of_sight = true;
-        for (const auto &obstacle : obstacles)
+        Point p1               = convertCoordToPoint(coord1);
+        Point p2               = convertCoordToPoint(coord2);
+        bool has_line_of_sight = false;
+
+        if (p1 == p2)
+        {
+            has_line_of_sight = true;
+        }
+        else
         {
-            if (obstacle->intersects(seg))
+            has_line_of_sight = true;
+            Segment seg(convertCoordToPoint(coord1), convertCoordToPoint(coord2));
+            for (const auto &obstacle : obstacles)
             {
-                has_line_of_sight = false;
-                break;
+                if (obstacle->intersects(seg))
+                {
+                    has_line_of_sight = false;
+                    break;
+                }
```

Comment:
Since has_line_of_sight is immediately set to *true* at the start of each logical block here - can we initially set it to *true* and have a single if(p1 != p2) block?

Reply:
changed it